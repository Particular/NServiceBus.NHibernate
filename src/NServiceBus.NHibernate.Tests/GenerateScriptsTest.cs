using NHibernate.Dialect;

namespace NServiceBus.NHibernate.Tests
{
    using System;
    using System.Collections.Generic;
    using System.Runtime.CompilerServices;
    using System.Threading.Tasks;
    using NServiceBus;
    using NHibernate;
    using NUnit.Framework;

    [TestFixture]
    public class DDL
    {
        [Test]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void Outbox()
        {
            var script = ScriptGenerator<MsSql2012Dialect>.GenerateOutboxScript();
            TestApprover.Verify(script);
        }

        [Test]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void Subscriptions()
        {
            var script = ScriptGenerator<MsSql2012Dialect>.GenerateSubscriptionStoreScript();
            TestApprover.Verify(script);
        }

        [Test]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void Timeouts()
        {
            var script = ScriptGenerator<MsSql2012Dialect>.GenerateTimeoutStoreScript();
            TestApprover.Verify(script);
        }

        [Test]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void GatewayDeduplication()
        {
            var script = ScriptGenerator<MsSql2012Dialect>.GenerateGatewayDeduplicationStoreScript();
            TestApprover.Verify(script);
        }

        [Test]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void MySaga()
        {
            var script = ScriptGenerator<MsSql2012Dialect>.GenerateSagaScript<MySaga>();
            TestApprover.Verify(script);
        }
    }

    class MySaga : Saga<MySaga.SagaData>, IAmStartedByMessages<MyMessage>
    {
        protected override void ConfigureHowToFindSaga(SagaPropertyMapper<SagaData> mapper)
        {
            mapper.ConfigureMapping<MyMessage>(m => m.UniqueId).ToSaga(s => s.UniqueId);
        }

        public class SagaData : ContainSagaData
        {
            public virtual string UniqueId { get; set; }
            public virtual IList<CollectionEntry> Entries { get; set; }
        }

        public class CollectionEntry
        {
            public virtual Guid Id { get; set; }
            public virtual decimal Value { get; set; }
        }

        public Task Handle(MyMessage message, IMessageHandlerContext context)
        {
            throw new NotImplementedException();
        }
    }

    class MyMessage : IMessage
    {
        public string UniqueId { get; set; }
    }
}
