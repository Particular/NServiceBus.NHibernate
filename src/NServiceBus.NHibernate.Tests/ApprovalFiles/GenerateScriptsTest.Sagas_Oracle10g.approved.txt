
    
    drop table MySaga_CollectionEntry cascade constraints
    
    drop table MySaga cascade constraints
    
    drop table EntriesWithoutId cascade constraints
    
    create table MySaga_CollectionEntry (
        Id RAW(16) not null,
       Value NUMBER(19,5),
       SagaData_id RAW(16),
       primary key (Id)
    )
    
    create table MySaga (
        Id RAW(16) not null,
       Originator VARCHAR2(255),
       OriginalMessageId VARCHAR2(255),
       UniqueId VARCHAR2(255) unique,
       primary key (Id)
    )
    
    create table EntriesWithoutId (
        SagaData_id RAW(16) not null,
       Value1 NUMBER(19,5),
       Value2 NUMBER(19,5)
    )
    
    create index IDX_BAE15720 on MySaga_CollectionEntry (SagaData_id)
    
    alter table MySaga_CollectionEntry 
        add constraint FK_BAE15720 
        foreign key (SagaData_id) 
        references MySaga
    
    create index IDX_AED12FB9 on EntriesWithoutId (SagaData_id)
    
    alter table EntriesWithoutId 
        add constraint FK_AED12FB9 
        foreign key (SagaData_id) 
        references MySaga